diff -uNr fuse-utils-1.0.0.old/configure.in fuse-utils-1.0.0/configure.in
--- fuse-utils-1.0.0.old/configure.in	2010-12-16 22:47:12.000000000 +0100
+++ fuse-utils-1.0.0/configure.in	2012-08-12 23:15:23.178640396 +0200
@@ -125,11 +125,18 @@
 audiofile=yes)
 AC_MSG_RESULT($audiofile)
 if test "$audiofile" = yes; then
-  AM_PATH_AUDIOFILE(
-    0.2.3,
-    AC_DEFINE([HAVE_LIB_AUDIOFILE], 1, [Defined if we've got audiofile]),
-    audiofile=no
+  PKG_CHECK_MODULES(
+    AUDIOFILE,
+    audiofile,
+    AC_DEFINE([HAVE_LIB_AUDIOFILE], 1, [Defined if we've got audiofile]), []
   )
+  if test -z "$AUDIOFILE_LIBS"; then
+    AM_PATH_AUDIOFILE(
+      0.2.3,
+      AC_DEFINE([HAVE_LIB_AUDIOFILE], 1, [Defined if we've got audiofile]),
+      audiofile=no
+    )
+  fi
 fi
 AM_CONDITIONAL(BUILD_AUDIOTOOLS, test "$audiofile" = yes)
 
